schema/encoding/jsonschema/benchmark_test.go

```diff
@@ -38,15 +39,25 @@ func BenchmarkEncoder(b *testing.B) {
 			},
 		},
 		{
-			Name:   `Schema=Student`,
+			Name:   `Schema=Simple`,
 			Schema: studentSchema,
 		},
+		{
+			Name:   `Schema=Complex1`,
+			Schema: requestSchema,
+		},
+		{
+			Name:   `Schema=Complex2`,
+			Schema: responseSchema,
+		},
 	}
+	buf := bytes.NewBuffer(make([]byte, 65535))
```

Comment:
Sorry, but why would anything call grow? we control the buffer and we only pass it in as an io.Writer. I like the Discard idea, but it could make debugging harder if we needed to pin point any problems. Discard would also not trigger any of the io.Writer error behavior since it's essentially a null instance. Truncate has the behavior I'm after. It's resets the buffer without doing any new allocations.

```
    67	func (b *Buffer) Truncate(n int) {
    68		b.lastRead = opInvalid
    69		switch {
    70		case n < 0 || n > b.Len():
    71			panic("bytes.Buffer: truncation out of range")
    72		case n == 0:
    73			// Reuse buffer space.
    74			b.off = 0
    75		}
    76		b.buf = b.buf[0 : b.off+n]
    77	}
```